lib/OpenQA/Worker/Cache.pm

```diff
@@ -151,18 +151,23 @@ sub get_asset {
     return $asset;
 }
 
-sub purge_cache {
-    log_debug("Expiring all the assets");
-    map { expire_asset($_) } @{$cache->{$host}};
-    write_db();
-}
-
 sub expire_asset {
     # currently only
     while (@{$cache->{$host}} > $limit) {
+        my $count = @{$cache->{$host}};
         my $asset = pop(@{$cache->{$host}});
-        unlink($asset) if $remove;
-        log_debug("Purged $asset due to $limit");
+        if (-e $asset) {
+            unlink($asset);
+            if ($@) {
+                log_error("Cannot purge $asset: $@");
```

Comment:
possibly if 2 workers cleanup at the same time. The locking between 2 workers is rather non-existant